clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -674,22 +675,19 @@ public void wakeup() {
         this.client.wakeup();
     }
 
-    public static Sensor throttleTimeSensor(Metrics metrics) {
-        String metricGrpName = SenderMetrics.METRIC_GROUP_NAME;
+    public static Sensor throttleTimeSensor(Metrics metrics, SenderMetricsRegistry metricsRegistry) {
```

Comment:
This is not a comment for the PR, but just a question for my own knowledge to @ijuma : this throttle time sensor is added to the network client layer, but we make it inside `Sender`. Why is this design?

Reply:
I agree that this is a bit weird and I asked the same question. If I remember correctly, it's done this way to record the values in a generic way, but the names are only known by the callers (depending on whether it's a producer or consumer). @rajinisivaram can elaborate.

Reply:
Yes, the sensors are created in `Sender/Fetcher` to avoid knowledge of the different names to the network layer. Recording is done in the network layer since `Sender/Fetcher` don't see all responses (now that any response may be throttled) and to use common logic.